README.md

```diff
@@ -3,3 +3,104 @@
 Knative is a Kubernetes-based platform to deploy and manage modern serverless workloads.
 
 For more details of what is possible with Knative, see https://knative.dev/
+
+## Usage
+
+### Pre-requisites
+
+Microk8s 1.21/stable
+>NOTE: These instructions assume you have run `microk8s enable dns storage rbac metallb:"10.64.140.43-10.64.140.49,192.168.0.105-192.168.0.111"`
+
+### Add a model and set variables
+
+```
+MODEL_NAME="knative"
+DEFAULT_GATEWAY="knative-gateway"
+juju add-model ${MODEL_NAME}
+```
+
+### Deploy dependencies
+
+Knative (in particular `knative-serving`) requires `istio-operators` to be deployed in the model. To correctly configure them, you can:
+
+```
+juju deploy istio-pilot --config default-gateway=${DEFAULT_GATEWAY} --trust
+juju deploy istio-gateway istio-ingressgateway --config kind="ingress" --trust
+juju relate istio-pilot istio-ingressgateway
+```
+
+### `knative-operator`
+
+The `knative-operator` is responsible for installing, configuring, and managing the `serving` and `eventing` components. Despite your final application, you must deploy the charmed `knative-operator` before any other Knative component.
```

Comment:
I think you're looking for a different word than "despite".  That kinda means "it doesn't affect your final application, but you must ...".  I don't think you need that part of the phrase anyway, but if you still want it you could say "Regardless of your final application, ..." or "In spite of ..."

```suggestion
The `knative-operator` is responsible for installing, configuring, and managing the `serving` and `eventing` components. You must deploy the charmed `knative-operator` before any other Knative component.
```